File: src/test/java/recipestore/db/triplestore/AbstractTripleStoreDAOTest.java
package recipestore.db.triplestore;

import org.apache.jena.rdf.model.Resource;
import org.hamcrest.Matchers;
import org.jooq.lambda.Seq;
import org.junit.After;
import org.junit.Test;

import java.util.List;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.greaterThan;

public abstract class AbstractTripleStoreDAOTest {

    abstract TripleStoreDAO getTripleStoreDAO();

    @Test
    public void shouldGetModelAndListStatements() {

        final List<Resource> statements = Seq.seq(getTripleStoreDAO().getResource("http://schema.org/Recipe")).toList();
        assertThat("Should have statements in triplestore", statements.size() > 0, Matchers.equalTo(true));

        assertThat("Should get all the recipe statements ", statements.size(), greaterThan(10));

    }


    @After
    public void cleanup() {
        getTripleStoreDAO().delete(true);
    }
}

File: src/test/java/recipestore/db/triplestore/AbstractTripleStoreDAOTest.scala
package recipestore.db.triplestore


import org.apache.jena.rdf.model.Resource
import org.scalatest.{BeforeAndAfterAll, FunSuite, Matchers}

import scala.collection.JavaConverters._

abstract class AbstractTripleStoreDAOTest extends FunSuite with Matchers with BeforeAndAfterAll {
  def getTripleStoreDAO: TripleStoreDAO

  def shouldGetModelAndListStatements() {
    val statements: List[Resource] = getTripleStoreDAO.getResource("http://schema.org/Recipe").asScala.toList
    statements.size > 0 should be(true)
    statements.size should be > 10
  }

  override def afterAll(): Unit = {
    getTripleStoreDAO.delete(true)
  }
}